### ISSUE TITLE
[Cleanup] remove internal package

### ISSUE THREAD

**Issue by jhendrixMSFT** [AzureAD/microsoft-authentication-library-for-go#57]
Internal packages are useful when you want to share, without exporting, code across several packages.  With the current implementation, the code is needlessly complex and hard to follow.

**Comment by henrik-me** [comment]
As long as API surface stays minimal and only what an MSAL would expose. Also should be considered if this change may impact the extensions library (I do not think so as it only leverages the public api, however we should ensure that is the case)

**Comment by element-of-surprise** [comment]
We now certainly are using internal/, but in a different way.  

Do you feel that we can close this now?

**Comment by jhendrixMSFT** [comment]
The original problems with internal have been resolved.
